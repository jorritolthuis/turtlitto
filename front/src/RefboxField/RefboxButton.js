import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * A button part of the refbox. Does a callback to its parent to handle onClick events
 * Author: S.A. Tanja
 * Author: G.W. van der Heijden
 *
 * Props:
 *  - teamColor: the team color of the button (cyan or magenta).
 *  - onClick: a function on what to do when the button is pressed
 *  - children: the children of the button
 */
const RefboxButton = props => {
  return (
    <Button
      className={props.className}
      teamColor={props.teamColor}
      onClick={() => {
        props.onClick();
      }}
    >
      {props.children}
    </Button>
  );
};

const Button = styled.button`
  background-color: ${props =>
    props.teamColor === "magenta"
      ? props.theme.refboxMagenta
      : props.theme.refboxCyan};
  justify-content: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid;
  user-select: none;
`;

RefboxButton.propType = {
  teamColor: PropTypes.oneOf(["cyan", "magenta"]).isRequired,
  tag: PropTypes.oneOf([
    "KO",
    "FK",
    "GK",
    "TI",
    "C",
    "P",
    "Soft",
    "Medium",
    "Hard"
  ]).isRequired
};

export default RefboxButton;
