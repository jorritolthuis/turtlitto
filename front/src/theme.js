// these colors are exposed by the styled-components ThemeProvider
// and used within the application
export default {
  success: "#00ff00",
  successLighter: "rgba(153, 255, 153, 0.25)",
  error: "#FF0000",
  errorLighter: "rgba(255,153,153, 0.4)",
  background: "linear-gradient(270deg, #196ebd, #01b0dd)",
  baseButton: "#FFFFFF",
  baseButtonDisabled: "rgba(216,216,216, 0.5)",
  info: "#bdbdbd",
  title: "#FFFFFF",
  secondary: "#d3d3d3",
  border: "#ccc",
  shadow: "rgba(0, 0, 0, 0.17)",
  turtleEnableBar: "#ededed",
  turtleEnableButton: "#ededed",
  turtleEnableButtonActive: "silver",
  bottomBarButton: "#ededed",
  bottomBarButtonActive: "silver",
  notificationSuccess: "green",
  notificationWarning: "#ffcc00",
  notificationError: "Tomato",
  dropBallButton: "#ededed",
  dropBallButtonActive: "silver",
  inOutButton: "#ededed",
  inOutButtonActive: "silver",
  refboxMagenta: "magenta",
  refboxCyan: "cyan",
  refboxMagentaActive: "magenta",
  refboxCyanActive: "cyan"
};
