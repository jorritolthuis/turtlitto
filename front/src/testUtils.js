import React from "react";
import { ThemeProvider } from "styled-components";
import theme from "./theme";
import { shallow, mount } from "enzyme";

export const shallowWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  });
};
