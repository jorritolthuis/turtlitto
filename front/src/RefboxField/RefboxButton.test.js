import React from "react";
import RefboxButton from "./RefboxButton";
import { shallowWithTheme } from "../testUtils";
import theme from "../theme";

describe("RefboxButton", () => {
  describe("is cyan", () => {
    it("should match snapshot", () => {
      const wrapper = shallowWithTheme(
        <RefboxButton teamColor="cyan" onClick={() => {}} tag="KO" />
      ).dive();
      expect(wrapper).toMatchSnapshot();
      expect(wrapper).toHaveStyleRule("background-color", theme.refboxCyan);
      expect(wrapper).toHaveStyleRule("border-color", theme.refboxCyanBorder);
      expect(wrapper).toHaveStyleRule("border", "0.2rem solid");
      expect(wrapper).toHaveStyleRule(
        "background-color",
        theme.refboxCyanActive,
        { modifier: ":active" }
      );
    });
  });

  describe("is magenta", () => {
    it("should match snapshot", () => {
      const wrapper = shallowWithTheme(
        <RefboxButton teamColor="magenta" onClick={() => {}} tag="P" />
      ).dive();
      expect(wrapper).toMatchSnapshot();
      expect(wrapper).toHaveStyleRule("background-color", theme.refboxMagenta);
      expect(wrapper).toHaveStyleRule(
        "border-color",
        theme.refboxMagentaBorder
      );
      expect(wrapper).toHaveStyleRule("border", "0.2rem solid");
      expect(wrapper).toHaveStyleRule(
        "background-color",
        theme.refboxMagentaActive,
        { modifier: ":active" }
      );
    });
  });
});
