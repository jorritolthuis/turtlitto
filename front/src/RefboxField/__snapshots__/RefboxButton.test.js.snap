// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefboxButton shows a cyan button with KO on it 1`] = `
.c0 {
  background-color: #00ffff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid #00c8c8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:active {
  background-color: #00c8c8;
}

<button
  className="c0"
  onClick={[Function]}
/>
`;

exports[`RefboxButton shows a magenta button with P on it 1`] = `
.c0 {
  background-color: #ff00ff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid #c800c8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:active {
  background-color: #c800c8;
}

<button
  className="c0"
  onClick={[Function]}
/>
`;
